File: common/src/main/java/cc/turtl/chiselmon/feature/pc/icon/IconEntry.java
package cc.turtl.chiselmon.feature.pc.icon;

import cc.turtl.chiselmon.config.category.PCConfig;
import com.cobblemon.mod.common.pokemon.Pokemon;
import net.minecraft.resources.ResourceLocation;

import java.util.function.Predicate;

public record IconEntry(
        ResourceLocation resource,
        Predicate<PCConfig.IconConfig> configCheck,
        Predicate<Pokemon> pokemonCheck
) {
    public boolean shouldDisplay(PCConfig.IconConfig config, Pokemon pokemon) {
        return configCheck.test(config) && pokemonCheck.test(pokemon);
    }
}

File: common/src/main/kotlin/cc/turtl/chiselmon/client/feature/pc/icon/IconEntry.kt
package cc.turtl.chiselmon.client.feature.pc.icon

import cc.turtl.chiselmon.client.config.category.PCConfig.IconConfig
import com.cobblemon.mod.common.pokemon.Pokemon
import net.minecraft.resources.ResourceLocation
import java.util.function.Predicate

data class IconEntry(
    val resource: ResourceLocation,
    val configCheck: Predicate<IconConfig>,
    val pokemonCheck: Predicate<Pokemon>
) {
    fun shouldDisplay(config: IconConfig, pokemon: Pokemon): Boolean {
        return configCheck.test(config) && pokemonCheck.test(pokemon)
    }
}